example/bench/bench.cpp

```diff
@@ -33,16 +35,26 @@ int main(int argc, char** argv) {
     bool is_root = true;
 
     try {
+        arb::proc_allocation resources;
+        if (auto nt = sup::get_env_num_threads()) {
+            resources.num_threads = *nt;
+        }
+        else {
+            resources.num_threads = sup::thread_concurrency();
+        }
+
 #ifdef ARB_MPI_ENABLED
         sup::with_mpi guard(argc, argv, false);
-        auto context = arb::make_context(arb::proc_allocation(), MPI_COMM_WORLD);
+        resources.gpu_id = sup::find_gpu(MPI_COMM_WORLD);
+        auto context = arb::make_context(resources, MPI_COMM_WORLD);
         {
-            int rank = 0;
+            int rank;
             MPI_Comm_rank(MPI_COMM_WORLD, &rank);
             is_root = rank==0;
         }
 #else
-        auto context = arb::make_context();
+        resources.gpu_id = sup::find_gpu();
```

Comment:
I think `find_gpu()` is not a good name; more on this below.

Reply:
Yep, I agree with the proposed changes below.